spectator/writer/unix_writer.py

```diff
@@ -9,10 +10,16 @@ class UnixWriter(Writer):
     def __init__(self, location: str) -> None:
         super().__init__()
         self._logger.debug("initialize UnixWriter to %s", location)
-        self._sock = socket.socket(family=socket.AF_UNIX, type=socket.SOCK_DGRAM)
         self._location = location
+        self._lock = threading.Lock()
+        self._sock = None
 
     def write(self, line: str) -> None:
+        if self._sock is None:
+            with self._lock:
+                if self._sock is None:
+                    self._sock = socket.socket(family=socket.AF_UNIX, type=socket.SOCK_DGRAM)
```

Comment:
In the case socket creation fails, it may be good to have this in the try block.  I believe spectator usually goes out of its way to avoid throwing exception in usual call paths